### PR TITLE
Build rate limit service config

### PR DESCRIPTION
**Pull request by arkodg**
* Translates Xds IR into ratelimit service config https://github.com/envoyproxy/ratelimit#configuration

Relates to https://github.com/envoyproxy/gateway/issues/670

Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

**Review by danehans — COMMENTED** [review]
@arkodg LGTM after rebasing.

File: internal/xds/translator/translator_test.go [hunk 1]
```diff
@@ -11,14 +11,15 @@ import (
...
 	"testing"
 
+	"github.com/envoyproxy/gateway/internal/ir"
 	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
 	resource "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	ratelimitserviceconfig "github.com/envoyproxy/ratelimit/src/config"
 	"github.com/stretchr/testify/require"
 	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/proto"
+	goyaml "gopkg.in/yaml.v3"
```
Comment (zirain) [inline internal/xds/translator/translator_test.go:21]: use `sigs.k8s.io/yaml` instead

**Comment by danehans** [comment]
@arkodg LGTM after you resolve https://github.com/envoyproxy/gateway/pull/893#discussion_r1070948747.

File: internal/xds/translator/translator_test.go (hunk 1 above)
Comment (arkodg) [reply internal/xds/translator/translator_test.go:21]: tried, doesnt work with it, the tags are `yaml` tags, which is why I add to add this package

Comment (zirain) [reply internal/xds/translator/translator_test.go:20]: merge this first, I will review depguard settings.

**Review by zirain — APPROVED** [review]

**Pull request merged**